Annotators
===============

Annotators can be divided into 2 groups:

1) Tools that predict pathogenicity (bold)
2) Tools that provide information like databases

• **cadd_exome** (1.6.1) CADD is a tool for scoring the deleteriousness of single nucleotide variants as well as insertion/deletions variants in the human genome.

• gnomad_gene (2.2.1) Gene level population statistics from gnomAD

• pubmed (1.1.5) articles related to a particular gene

• **clingen** (1.0.1) - NIH-funded resource that defines the clinical relevance of genes and variants

• **clinpred** (1.0.0) - prediction tool to identify disease-relevant nonsynonymous single nucleotide variants

• clinvar (2021.10.01) - ClinVar is an archive of reports of the relationships among human variations and phenotypes, archive of interpretations of clinically relevant variants (Uncertain significance, Likely pathogenic, Pathogenic etc.)

• mitomap (1.1.0) A human mitochondrial genome database

• ncbigene (2019.08.02) -  gene descriptions from NCBI (National Center for Biotechnology Information)Gene database.

• omim (1.0.0) Catalog of human genes and genetic disorders and traits.

• **prec** (3.6.0)provides a database identifying rare and likely deleterious loss-of-function (LoF) alleles

• **provean** (1.0.0), tool which predicts whether an amino acid substitution or indel has an impact on the biological function of a protein

• **revel** (2020.12.02), ensemble method for predicting the pathogenicity of missense variants based on a combination of scores from 13 individual tools

• **sift** (1.2.0) predicts whether an amino acid substitution affects protein function based on sequence homology and the physical properties of amino acids

• GnomADD aggregating and harmonizing both exome and genome sequencing data from a wide variety of large-scale sequencing projects

• PharmGKB

• dbSNP
